src/main/java/Duke.java

```diff
@@ -1,10 +1,39 @@
+import java.util.Scanner;
+
 public class Duke {
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+        UI.showWelcomeScreen();
+
+        Scanner sc = new Scanner(System.in);
+
+        Task[] tasks = new Task[100];
+        String input;
+
+        do {
+            input = sc.nextLine();
+            switch (input) {
+            case "list":
+                System.out.println("This is your current list:");
+                for (int i = 0; i < Task.getTaskCounter(); i++){
+                    int displayedTask = i + 1;
+                    System.out.println(displayedTask + ": " + tasks[i]);
+                }
+                UI.showDivider();
+                break;
+            case "bye":
+                System.out.println(UI.divider);
+                System.out.println("Bye! Hope to hear from you again soon!");
+                System.out.println(UI.divider);
+                break;
+            default:
+                // todo make the heavy lifting be done by another processing class instead of here
+                Parser.addOrCompleteTask(input, tasks);
+
+            }
+        } while (!input.equals("bye"));
```

Comment:
Varying levels of abstraction.

The abstraction level of this entire block of code, when compared with "UI.showWelcomeScreen();" at line 7, has different levels of abstraction. Perhaps you may want to shift the switch statement block of code to a separate method, so as to achieve "Single Level of Abstraction Principle".

Reply:
Ah I understand - thank you!